### PR TITLE
Add DataWarehouseRole in template

### PR DESCRIPTION
**Pull request by monaullah**
See cross-account permissions and using IAM roles: https://docs.aws.amazon.com/AmazonS3/latest/userguide/example-walkthroughs-managing-access-example4.html#access-policies-walkthrough-example4-overview

Data warehouse account must provide external ID to assume role, the primary function of the external ID is to address the [confused deputy problem](http://en.wikipedia.org/wiki/Confused_deputy_problem) (from https://aws.amazon.com/blogs/apn/securely-using-external-id-for-accessing-aws-accounts-owned-by-others/)

### CODE REVIEW

File: template.yaml [hunk 1]
```diff
@@ -243,6 +264,22 @@ Resources:
...
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              AWS: "arn:aws:iam::${DataWarehouseAccountId}:root"
+            Action: sts:AssumeRole
+            Condition:
+              StringEquals:
+                sts:ExternalId: !Ref DataWarehouseExternalId
```
Comment (sondrev) [inline template.yaml:280]: Can you explain what this does?

Comment (monaullah) [reply template.yaml:267]: It's an extra security measure, when someone from the other aws account assumes this role via the aws Security Token Service API (sts), they must provide an external id.

See linked docs:  https://aws.amazon.com/blogs/apn/securely-using-external-id-for-accessing-aws-accounts-owned-by-others/

Comment (monaullah) [reply template.yaml:267]: It's to prevent other accounts from assuming this role via data warehouse account.

Comment (brinxmat) [reply template.yaml:267]: Can we add a description to the role to say that this provides external access when paired with known external account and Id?

**Review by brinxmat — APPROVED** [review]

**Review by torbjokv — APPROVED** [review]

**Review by brinxmat — APPROVED** [review]

**Comment by torbjokv** [comment]
/NVA/IAM/DataWarehouse/DataBricksArn and ExternalId should be defined in nva-infrastructure?

**Pull request merged**